ext/opcache/Optimizer/zend_dump.c

```diff
@@ -554,7 +556,7 @@ static void zend_dump_op(const zend_op_array *op_array, const zend_basic_block *
 	} else if (ZEND_VM_EXT_SRC == (flags & ZEND_VM_EXT_MASK)) {
 		if (opline->extended_value == ZEND_RETURNS_VALUE) {
 			fprintf(stderr, " (value)");
-		} else if (opline->extended_value == ZEND_RETURNS_FUNCTION) {
+		} else if (opline->extended_value & ZEND_RETURNS_FUNCTION) {
```

Comment:
There is some mess with flags:
```php
<?php
class A {
	public static ?int $a;
}
class B {
	public static ?float $b;
}

$x = & A::$a;
B::$b = & $x;
```
L0 (9):     V1 = FETCH_STATIC_PROP_W 0 (packed) (ref) string("a") string("A")
L1 (9):     ASSIGN_REF CV0($x) V1
L2 (10):    ASSIGN_STATIC_PROP_REF (function) string("b") string("B")
L3 (10):    OP_DATA CV0($x)
L4 (13):    RETURN int(1)
```

```